### PR TITLE
Added citext parsing tests

### PR DESCRIPTION
**Pull request by taion809**
This pull request adds `citext` specific tests to the `CustomTypesSpec` integration tests and `ValuesSpec` tests

### CODE REVIEW

**Comment by jeremyrsmith** [comment]
@taion809 Looks like the tests are failing because the string is being lowercased, so it's not the same string anymore. Can you fix the test for citext to handle this?

File: src/test/scala/com/twitter/finagle/postgres/integration/CustomTypesSpec.scala [hunk 1]
```diff
@@ -105,6 +103,13 @@ class CustomTypesSpec extends Spec with GeneratorDrivenPropertyChecks {
       } catch {
         case err: Throwable => // can't run this one because we're not superuser
       }
+    
+      try {
+        Await.result(client.query("CREATE EXTENSION IF NOT EXISTS citext"))
+        "parse citext" in test(ValueDecoder.String)("citext")
```
Comment (jeremyrsmith) [inline src/test/scala/com/twitter/finagle/postgres/integration/CustomTypesSpec.scala]: Could do `test(ValueDecoder.String)("citext", (a, b) => a.toLowerCase == b.toLowerCase)`

**Comment by njohns-grovo** [comment]
no luck :(

```
[info] - must parse citext *** FAILED ***
[info]   ClassCastException was thrown during property evaluation.
[info]     Message: [B cannot be cast to java.lang.String
[info]     Occurred when passed generated values (
[info]       arg0 = ""
[info]     )
```

**Comment by taion809** [comment]
no luck :(

```
[info] - must parse citext *** FAILED ***
[info]   ClassCastException was thrown during property evaluation.
[info]     Message: [B cannot be cast to java.lang.String
[info]     Occurred when passed generated values (
[info]       arg0 = ""
[info]     )
```

I had posted this into the gitter channel but there doesn't appear to be a `citextrecv` function in the postgresql `citext.c` source code

**Comment by jeremyrsmith** [comment]
I'm going to close this for now. I *think* citext works properly, but if it doesn't then we'd need to think of a good story for handling extensions with custom types.

**Pull request closed without merging**